package/MDAnalysis/core/groups.py

```
@@ -3324,6 +3324,66 @@ def write(self, filename=None, file_format=None,
 
         raise ValueError("No writer found for format: {}".format(filename))
 
+    def sort(self, key='ix', keyfunc=None):
+        """Return stably sorted Atomgroup by a key that specifies
+           an attribute of atomgroup. The default key is "ix".
+
+        Parameters
+        ----------
+        keys: str
+            The name for attribute of Atomgroup. (e.g. "ids", "ix")
+        keyfunc: function
+            A function that returns 1 dimension array, a key for sorting the
+            atomgroup from multiple dimension array that the attribute returns.
+
+        Returns
+        -------
+        output:
+            Sorted atomgroup.
+
+        Example
+        -------
+
+        .. code-block:: python
+
+            >>> import MDAnalysis as mda
+            >>> from MDAnalysisTests.datafiles import PDB_small
+            >>> u = mda.Universe(PDB_small)
+            >>> ag = u.atoms[[3, 2, 1, 0]]
+            >>> ag.ids
+            array([4 3 2 1])
+            >>> ag.ix
+            array([3 2 1 0])
+            >>> ag = ag.sort()
+            >>> ag.ids
+            array([1 2 3 4])
+            >>> ag.ix
+            array([0 1 2 3])
+
+        .. versionadded:: 2.0.0
+        """
+        idx = getattr(self.atoms, key)
+        if len(idx) != len(self.atoms):
+            raise ValueError("The array returned by the attribute '{}' "
+                             "must have a same length as the number of "
+                             "atoms".format(key))
+        if idx.ndim == 1:
+            order = np.argsort(idx, kind='stable')
+        elif idx.ndim > 1:
```

I'm not fully convinced this is in the scope of what we want to cover with this function, but I'll defer to @richardjgowers' opinion though.

---

I thought someone may want to sort by extracts of multiple dimensional arrays like the x-axis of positions. However, I also feel it should just raise an error when `idx.ndim` is not equal to 1.